src/time-to-interactive/TTIBoomerang.ts

```diff
@@ -0,0 +1,331 @@
+                        this.ttiTracker[LONG_TASK][bucket] >
+                            this.LONG_TASK_THRESHOLD
+                    ) {
+                        allOk = false;
+                    }
+
+                    // Check FPS
+                    if (
+                        this.fpsSupported &&
+                        this.ttiTracker[FPS] &&
+                        this.ttiTracker[FPS][bucket] &&
+                        this.ttiTracker[FPS][bucket] < this.FPS_THRESHOLD
+                    ) {
+                        allOk = false;
+                    }
+
+                    if (allOk) {
+                        okIntervals += 1;
+                        if (okIntervals >= this.REQUIRED_OK_INTERVALS) {
+                            this.ttiResolved = true;
+                            clearInterval(tti);
+                            // TTI is Visually Ready + time to start of ok period from visually ready
+                            const timeToQuietPeriodFromVisuallyReady: number =
+                                (currBucket -
+                                    this.REQUIRED_OK_INTERVALS -
+                                    startBucket +
+                                    1) *
+                                this.COLLECTION_PERIOD;
+                            resolve(
+                                visuallyReadyTimestamp +
+                                    timeToQuietPeriodFromVisuallyReady
+                            );
+                            break;
+                        } else {
+                            continue;
+                        }
+                    } else {
+                        okIntervals = 0;
+                    }
+
+                    currBucket = endBucket;
+                }
+                totalIntervals += 1;
+            }, this.CHECK_PERIOD);
+        });
+    }
+
+    initListeners(): void {
+        // record domcontentloaded if its found already
+        if (window.performance) {
+            if (window.performance.getEntriesByType(NAVIGATION)) {
+                this.domContentLoadedEventEnd =
+                    window.performance.getEntriesByType(
+                        NAVIGATION
+                    )[0]?.domContentLoadedEventEnd;
+            }
+        }
+
+        // Use performance observer to record long tasks and domcontentloaded
+        this.eventListener();
+
+        // Record support for FCP and LCP and init listeners
+        this.lcpSupported = isLCPSupported();
+        this.fcpSupported = isFCPSupported();
+
+        // Use web vitals library for LCP and FCP, if supported
+        if (this.fcpSupported) {
+            onFCP((metric) => this.handleWebVitals(metric));
+        }
+        if (this.lcpSupported) {
+            onLCP((metric) => this.handleWebVitals(metric));
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        if (window.requestAnimationFrame(() => {})) {
```

Comment:
Question: what is the purpose this this call and when would it return a falsy value (0)? Documentation states `This is a non-zero value, but you may not make any other assumptions about its value.`

https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame#return_value

Reply:
I see, if we are just checking for support for requestAnimationFrame, then should this just check that the function exists? 
```suggestion
        if (window.requestAnimationFrame) {
```